### PR TITLE
Portable PDB + Offline Store Support

### PR DESCRIPTION
**Pull request by xenolightning**

### CODE REVIEW

File: Raygun4Maui/Platforms/Android/AndroidUtilities.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        // Open the apk file, and see if it has a manifest, if it does,
+        // we are using the new assembly store method,
+        // else it's just a normal zip with assemblies as archive entries
+        using var zipArchive = ZipFile.Open(apkPath, ZipArchiveMode.Read);
+
+        foreach(var entry in zipArchive.Entries)
+        {
+            Console.WriteLine(entry.FullName);
+        }
+
```
Comment (ProRedCat) [inline Raygun4Maui/Platforms/Android/AndroidUtilities.cs:47]: Console outs

File: Raygun4Maui/Platforms/Android/AndroidUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        if (zipArchive.GetEntry("assemblies/assemblies.manifest") != null)
+        {
+            Console.WriteLine("Creating Blob");
```
Comment (ProRedCat) [inline Raygun4Maui/Platforms/Android/AndroidUtilities.cs:50]: Another

File: Raygun4Maui/Platforms/Android/AssemblyBlobStoreReader.cs [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+                }
+            });
+
+            ProcessIndex(_indexStore.GlobalIndex64, (he, assembly) =>
+            {
+                assembly.Hash64 = he.Hash;
+                if (assembly.RuntimeIndex != he.MappingIndex)
+                {
+                    Debug.WriteLine($"assembly with hashes 0x{assembly.Hash32} and 0x{assembly.Hash64} has a different 32-bit runtime index ({assembly.RuntimeIndex}) than the 64-bit runtime index({he.MappingIndex})");
+                }
```
Comment (ProRedCat) [inline Raygun4Maui/Platforms/Android/AssemblyBlobStoreReader.cs:148]: Debug writeline - do we want to keep this?

File: Raygun4Maui/Platforms/Android/AssemblyBlobStoreReader.cs [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+                        if (string.IsNullOrEmpty(assembly.Name))
+                        {
+                            Debug.WriteLine($"64-bit hash 0x{assembly.Hash64:x} did not match any assembly name in the manifest");
+                        }
+                    }
+                    else if (!string.Equals(assembly.Name, me.Name, StringComparison.Ordinal))
+                    {
+                        Debug.WriteLine($"32-bit hash 0x{assembly.Hash32:x} maps to assembly name '{assembly.Name}', however 64-bit hash 0x{assembly.Hash64:x} for the same entry matches assembly name '{me.Name}'");
+                    }
+                }
```
Comment (ProRedCat) [inline Raygun4Maui/Platforms/Android/AssemblyBlobStoreReader.cs:165]: More debug

File: Raygun4Maui/Platforms/Android/AssemblyBlobStoreReader.cs (hunk 3 above)
Comment (xenolightning) [reply Raygun4Maui/Platforms/Android/AssemblyBlobStoreReader.cs:148]: Yeah these are intentional

**Review by ProRedCat — APPROVED** [review]
Lgtm, nice work on selecting the ABI specific package - that could have been a tricky bug

Just some further notes for visibility, this should all work but it's possible when an app is deployed to the app store some shenanigans take place which break this so we should keep an eye on that.

**Pull request merged**